code/game/atoms_movable.dm

```diff
@@ -160,10 +159,10 @@
 			if(destarea && old_area != destarea)
 				destarea.Entered(src, oldloc)
 
-		for(var/atom/movable/AM in destination)
-			if(AM == src)
-				continue
-			AM.Crossed(src)
+				for(var/atom/movable/AM in destination)

```

Comment:
https://github.com/HippieStation/HippieStation13/pull/183/files/9a9872a3f1f280eeb780e45bb612875bb31c455a#diff-85b909bea63591ebec264bda09fbb017L142 look at this